src/libs/Pusher/pusher.js

```diff
@@ -1,8 +1,21 @@
+import Onyx from 'react-native-onyx';
 import _ from 'underscore';
+import ONYXKEYS from '../../ONYXKEYS';
 import Pusher from './library';
 import TYPE from './EventType';
 import Log from '../Log';
 
+let isOffline = false;
+Onyx.connect({
+    key: ONYXKEYS.NETWORK,
+    callback: (network) => {
+        if (!network) {
+            return;
+        }
+        isOffline = Boolean(network.shouldForceOffline) || network.isOffline;
```

Comment:
I have a little concern about using `network.isOffline`, I feel like we could end up dropping push events if the connection is flickery. To be safe, I would only use `shouldForceOffline`.

Reply:
If the app is offline how would we get the Push event? I added this to solve the problem where Push events arrive while going offline with the dev tools. I think it's a worthwhile fix.

Reply:
My concerns comes from that sometimes the app doesn't seem to be 100% of the correct about it having a connection or not. I have seen it showing offline while I have a connection, and I guess it could also have some delay in updating this state. `network.isOffline` could be still saying that it is offline, but is really not and a pusher event arrive.

If you are convinced that this is not a problem, I can double check by sending lots of pusher events while enabling/disabling the connection and see if it actually happens or not

Reply:
Thanks for explaining a bit more. I haven't encountered a situation where the app thinks it's offline but it's actually online. Let's definitely test it thoroughly.

Reply:
Since this change is making the unit tests fail and you have suggested that it might be fragile I'll get rid of it.

Reply:
Oh actually something else was causing the tests to fail, but still, I think you're right that it's safer to only look at `shouldForceOffline`.

Reply:
Just confirming: We reverted the logic in this [commit](https://github.com/Expensify/App/pull/12135/commits/3104a7e3fbf81523d6b7b3d17f7d715bbcdb6920), so I should be receiving the pusher events when setting the network to offline with dev tools right? It's not the case right now. I am not receiving pusher events.

Reply:
Hmm interesting. That might have to do with your environment set up or using the web proxy. I don't think it's a blocker for this PR.